tests/crypto_test.py

```diff
@@ -26,3 +56,286 @@ def test_returned_string_is_lower_case(self):
     def test_raises_for_unknown_hash_name(self):
         with self.assertRaises(GPGProblem):
             crypto.RFC3156_micalg_from_algo(gpgme.MD_NONE)
+
+
+class _TestKey(utilities.TestCaseClassCleanup):
+
+    # A useful single fingerprint for tests that only care about one key. This
+    # key will not be ambiguous
+    FPR = "F74091D4133F87D56B5D343C1974EC55FBC2D660"
+
+    # Some additional keys, these keys may be ambigiuos
+    EXTRA_FPRS = [
+        "DD19862809A7573A74058FF255937AFBB156245D",
+        "2071E9C8DB4EF5466F4D233CF730DF92C4566CE7",
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        # Create a temporary directory to use as gnupg's home directory. This
+        # allows us to import keys without dirtying the user's keyring
+        home = tempfile.mkdtemp()
+        cls.addClassCleanup(shutil.rmtree, home)
+        mock_home = mock.patch.dict(os.environ, {'GNUPGHOME': home})
+        mock_home.start()
+        cls.addClassCleanup(mock_home.stop)
+
+        # create a single context to use class wide.
+        cls.ctx = gpgme.Context()
+        cls.ctx.armor = True
+
+        # Add the public and private keys. They have no password
+        search_dir = os.path.join(os.path.dirname(__file__), 'static/gpg-keys')
+        for each in os.listdir(search_dir):
+            if os.path.splitext(each)[1] == '.gpg':
+                with open(os.path.join(search_dir, each)) as f:
+                    cls.ctx.import_(f)
+
+        cls.key = cls.ctx.get_key(cls.FPR)
+
+
+class TestDetachedSignatureFor(_TestKey):
+
+    def test_valid_signature_generated(self):
+        to_sign = "this is some text.\nit is more than nothing.\n"
+        _, detached = crypto.detached_signature_for(to_sign, self.key)
+
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(detached)
+            sig = f.name
+        self.addCleanup(os.unlink, f.name)
+
+        with tempfile.NamedTemporaryFile(delete=False) as f:
+            f.write(to_sign)
+            text = f.name
+        self.addCleanup(os.unlink, f.name)
+
+        res = subprocess.check_call(['gpg', '--verify', sig, text])
```

Comment:
Yes, the same reason as the other case, I want to use a different mechanism to prevent hiding bugs through misconfiguration. I can certainly send output to /dev/null